### PR TITLE
Issue#127 additional patient identification + ssn

### PR DESCRIPTION
**Pull request by cragun47**
This adds / improves patient identifier function
1.  Multiple IDs
2. Uses System ID format rather than  assigner:reference
3. Uses SSN from PID.19 if it is present.

Adds tests
Updates naming convention for DataTypes that are slightly different from each other.

### CODE REVIEW

File: src/main/resources/hl7/datatype/Identifier_Gen.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ period:
     vars:
```
Comment (LisaWellman) [inline src/main/resources/hl7/datatype/Identifier_Gen.yml:20]: Can you add a comment to the top of this file, like you did in Identifier.yaml (I could not place this comment at the top like I wanted)

File: src/test/java/io/github/linuxforhealth/hl7/message/Hl7ORUMessageTest.java [hunk 2]
```diff
@@ -206,7 +206,7 @@ public void multipleOBXWithNoId() throws IOException {
       List<Resource> organizationResource = e.stream()
               .filter(v -> ResourceType.Organization == v.getResource().getResourceType())
               .map(BundleEntryComponent::getResource).collect(Collectors.toList());
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/message/Hl7ORUMessageTest.java:208]: I think this is worth a code comment stating that we do not expect an organization to be generated for patient identifier assigning authorities.

File: src/test/java/io/github/linuxforhealth/hl7/message/Hl7ORUMessageTest.java [hunk 3]
```diff
@@ -282,7 +282,7 @@ public void multipleOBXWithMixedType() throws IOException {
...
-      assertThat(organizationResource).hasSize(1);
+      assertThat(organizationResource).isEmpty();
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/message/Hl7ORUMessageTest.java:285]: I think this is worth a code comment stating that we do not expect an organization to be generated for patient identifier assigning authorities.

File: src/test/java/io/github/linuxforhealth/FHIRConverterTest.java [hunk 4]
```diff
@@ -246,7 +246,7 @@ public void test_VXU_V04_message() {
     List<Resource> organizationRes = e.stream()
         .filter(v -> ResourceType.Organization == v.getResource().getResourceType())
         .map(BundleEntryComponent::getResource).collect(Collectors.toList());
-    assertThat(organizationRes).hasSize(2);
+    assertThat(organizationRes).hasSize(1);
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/FHIRConverterTest.java:249]: I think this is worth a code comment stating that we do not expect an organization to be generated for patient identifier assigning authorities, but there will be an organization for the manufacturer.  Ideally this test would be verifying more than just a org count, but good with just improving the code comments for now!

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+import io.github.linuxforhealth.hl7.segments.util.PatientUtils;
+
+public class Hl7IdentifierFHIRConversionTest {
+
+  @Rule
+  public ExpectedException exceptionRule = ExpectedException.none();
+
+
+  @Test
+
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:28]: Remove extra whitespace

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    cc = identifier.getType();
+    assertThat(cc.getText()).hasToString("SS");
+    assertThat(cc.hasCoding()).isTrue(); 
+
+    // Third identifier (Driver's license) shallow check
+    identifier = identifiers.get(2);
+    assertThat(identifier.hasSystem()).isTrue();
+    assertThat(identifier.getSystem()).hasToString("urn:id:MNDOT");
+    assertThat(identifier.getValue()).hasToString("MN1234567");
+    assertThat(identifier.hasType()).isTrue();
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:72]: Make this a medium check, should have "DL" for the code

**Review by LisaWellman — APPROVED** [review]
Some small comments, approving.

File: src/test/java/io/github/linuxforhealth/FHIRConverterTest.java (hunk 4 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/FHIRConverterTest.java:249]: Will do.

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java (hunk 6 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7IdentifierFHIRConversionTest.java:72]: OK

File: src/main/resources/hl7/datatype/Identifier_Gen.yml (hunk 1 above)
Comment (cragun47) [reply src/main/resources/hl7/datatype/Identifier_Gen.yml:20]: Done

File: src/test/java/io/github/linuxforhealth/hl7/message/Hl7ORUMessageTest.java (hunk 3 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/message/Hl7ORUMessageTest.java:285]: Done

File: src/test/java/io/github/linuxforhealth/FHIRConverterTest.java (hunk 4 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/FHIRConverterTest.java:249]: Done

**Pull request merged**